### PR TITLE
Create version of tariff editor to support school group tariffs

### PR DESCRIPTION
**Pull request by iant**
These tariffs can only be created or updated by ES admins or group admins.

We won’t make this editor available to group admins initially, so links to the new pages should be in the ES admin only section of the Manage Group menu

- [x] The tariffs will be linked to the School Group as the tariff holder
- [x] Only gas and electricity meter types will be supported. So no tariffs tied to specific meters or support for solar/exported solar.
- [x] The tariffs can have an option start or end time, but not both
- [x] The entry page should list the tariffs for the school group, allowing them to viewed, edited, etc.
- [x] For this version of the editor there will be no “Smart Meter” tariffs tab.
- [x] The editor will have a “Default tariffs” tab. It should list the system wide tariffs i.e. owned by the SiteSettings

### CODE REVIEW

File: app/views/school_groups/_sub_nav.html.erb [hunk 1]
```diff
@@ -21,6 +21,7 @@
             <div class="dropdown-menu scrollable" aria-labelledby="manage-school-group" id="dropdown-manage-school-group">
               <%= link_to t('school_groups.sub_nav.chart_settings'), school_group_chart_updates_path(@school_group), class: 'dropdown-item' %>
               <%= link_to t('school_groups.sub_nav.clusters'), school_group_clusters_path(@school_group), class: 'dropdown-item' %>
+              <%= link_to t('manage_school_menu.manage_tariffs'), school_group_energy_tariffs_path(@school_group), class: 'dropdown-item' %>
```
Comment (ldodds) [inline app/views/school_groups/_sub_nav.html.erb:24]: "We won’t make this editor available to group admins initially, so links to the new pages should be in the ES admin only section of the Manage Group menu". Can you move this into the admin section?

If this impacts the tests, then you could comment those out for now. I want to allow us to internally test this out before making it visible to users.

File: app/services/energy_tariff_default_prices_creator.rb [hunk 2]
```diff
@@ -10,6 +10,7 @@ def process
 
     night_times = case @energy_tariff.tariff_holder_type
                   when 'School' then Economy7Times.times(@energy_tariff&.meters&.first&.mpan_mprn)
+                  when 'SchoolGroup' then TimeOfDay.new(0, 0)..TimeOfDay.new(7, 0) # Equivalent to Economy7Times::DEFAULT_TIMES
```
Comment (ldodds) [inline app/services/energy_tariff_default_prices_creator.rb:13]: Constant should be visible now in latest analytics. Could do this later as part of further improvements.

File: app/controllers/energy_tariffs/energy_tariffs_controller.rb [hunk 3]
```diff
@@ -4,23 +4,30 @@ class EnergyTariffsController < ApplicationController
...
+      authorize! :manage, @school.energy_tariffs.build if @school
+      authorize! :manage, @school_group.energy_tariffs.build if @school_group
+
       if @school
         @electricity_meters = @school.meters.electricity
         @electricity_tariffs = @school.energy_tariffs.electricity.by_start_date.by_name
         @gas_meters = @school.meters.gas
         @gas_tariffs = @school.energy_tariffs.gas.by_start_date.by_name
       end
+      @site_setting = SiteSettings.current if @school_group
```
Comment (ldodds) [inline app/controllers/energy_tariffs/energy_tariffs_controller.rb:23]: With this set, then we end up seeing duplicate tariffs on the school group page:

```
    <%= render 'energy_tariffs_tab_school_groups' if @school_group %>
    <%= render 'energy_tariffs_tab_site_settings' if @site_setting %>
```

Means we see the school groups tariffs, then all of the site settings. The latter should only be in the default view.

File: app/views/energy_tariffs/energy_tariffs/_energy_tariffs_summary_table.html.erb [hunk 4]
```diff
@@ -35,7 +35,7 @@
             <%= link_to t('common.labels.edit'), edit_school_energy_tariff_path(school, energy_tariff), class: 'btn' %>
             <%= link_to t('common.labels.delete'), school_energy_tariff_path(school, energy_tariff), method: :delete, data: {confirm: t('common.confirm') }, class: 'btn btn-danger' %>
           <% end %>
-          <% if energy_tariff.tariff_holder_type == 'SiteSettings' %>
+          <% if energy_tariff.tariff_holder_type == 'SiteSettings' && site_settings_page? %>
```
Comment (ldodds) [inline app/views/energy_tariffs/energy_tariffs/_energy_tariffs_summary_table.html.erb:38]: There's no if statement checking for whether the tariff holder is a SchoolGroup which means there's no "Full details", "Edit" or "Delete" button, so I can't do anything with a tariff once created.

**Review by ldodds — CHANGES_REQUESTED** [review]

File: app/views/school_groups/_sub_nav.html.erb (hunk 1 above)
Comment (iant) [reply app/views/school_groups/_sub_nav.html.erb]: done :+1: (see skipped test)

File: app/services/energy_tariff_default_prices_creator.rb (hunk 2 above)
Comment (iant) [reply app/services/energy_tariff_default_prices_creator.rb]: :+1:

File: app/controllers/energy_tariffs/energy_tariffs_controller.rb (hunk 3 above)
Comment (iant) [reply app/controllers/energy_tariffs/energy_tariffs_controller.rb]: :+1: done

**Review by ldodds — APPROVED** [review]

**Pull request merged**